### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by roshan-bajgain**
## Testing for music_album
- Add unit tests for all implemented methods

### CODE REVIEW

File: spec/music_album_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+describe MusicAlbum do
+  context 'It should create a musicAlbum instance with onspotify, name and published date' do
+    music = MusicAlbum.new(publish_date: '2012-05-01', name: true, on_spotify: 'Hello')
+    it 'creates new musicAlbum with the given parameters' do
+      expect(music).to be_an_instance_of MusicAlbum
+    end
+    it 'should check if it can be archived' do
+      expect(music.on_spotify).to eq 'Hello'
+    end
```
Comment (matovu-farid) [inline spec/music_album_spec.rb]: This does not check whether music is archived.
Please add the correct method to check whether the item is archived

Comment (matovu-farid) [reply spec/music_album_spec.rb]: You also need to check both cases, when the music album is archived and when it is not archived

File: spec/music_album_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative '../classes/music_album'
+
+describe MusicAlbum do
+  context 'It should create a musicAlbum instance with onspotify, name and published date' do
+    music = MusicAlbum.new(publish_date: '2012-05-01', name: true, on_spotify: 'Hello')
```
Comment (matovu-farid) [inline spec/music_album_spec.rb]: Please observe the correct argument types. on_spotify is a boolean, and publish_date is a Date from the DateTime class.
Incorrect arguments lead to invalid tests that don't simulate the behaviour of the code. This defeats the purpose of testing.

**Review by matovu-farid — CHANGES_REQUESTED** [review]
## Request changes ♻️ 
Hi @roshan-bajgain , 
I love the work you have put in so far in developing your project. 

- [x] All your tests pass
- [x] Tests that are implemented are meaningful
- [x] Tests are clearly labeled 
**However, there are a few changes to be made to make your project complete.**
_**Please check the comments below**_

**Comment by roshan-bajgain** [comment]
@matovu-farid  I have implemented all the changes according to you. kindly check it

**Review by matovu-farid — APPROVED** [review]
Hi @roshan-bajgain,
Your task is complete. There is nothing else to say other than it is time to merge it
✔️  No linter errors
✔️  All project requirements met

**Pull request merged**